api/prisma/reindex.ts

```diff
@@ -15,31 +15,51 @@ const reindex = async () => {
 
     const pubs = await client.prisma.publication.findMany({
         where: {
-            currentStatus: 'LIVE'
+            versions: {
+                some: {
+                    currentStatus: 'LIVE'
+                }
+            }
+        },
+        include: {
+            versions: {
+                where: {
+                    currentStatus: 'LIVE'
+                },
+                orderBy: {
+                    versionNumber: 'desc'
+                }
```

Comment:
That would exclude publications where there is a live version but also an active draft. It could look like this for example:
```
...
versions: {
    ...
    {
        ...
        versionNumber: 2,
        isCurrent: false,
        currentStatus: 'LIVE'
    },
    {
        ...
        versionNumber: 3,
        isCurrent: true,
        currentStatus: 'DRAFT'
    }
}
```
This wouldn't be returned if we used that where clause.

Reply:
That where clause I suggested only applies to versions, not to the publications. It filters the versions and only returns the ones where "currentStatus=LIVE" and "isCurrent=true" which means it will only return one version, instead of all of them in a reverse order. It should run faster.

Reply:
In the example you have provided the data is wrong. If we have v2 and v3, and v3 is a DRAFT, then v2 MUST be the latest LIVE version. Otherwise our system is broken, because it should only allow creating a new version from the latest LIVE version which will continue to be the latest until the new DRAFT becomes LIVE.

Reply:
I get it - I would do it that way as well, but it would still exclude versions (I should have said versions above, not publications) like version 2 in the example I put above. Because there is a new draft (which would have isCurrent = true, and only one version can have this), the latest live version no longer has isCurrent = true. But the latest live version is the one we want to put into the index. So we can't only include versions that have isCurrent = true.

Reply:
Closing this due to lack of info on how we are going to implement the new version creation mechanism. We may revisit it later on after we have an implementation of OC-391